DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs

```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
         [SetUp]
         public void Setup()
         {
-            _courseRepositoryMock = new Mock<ICourseRepository>();
             _topicRepositoryMock = new Mock<ITopicRepository>();
+            _courseRepositoryMock = new Mock<ICourseRepository>();
             _taskRepositoryMock = new Mock<ITaskRepository>();
             _materialRepositoryMock = new Mock<IMaterialRepository>();
         }
+
+        [Test]
+        public void AddCourse_CourseInput_CourseCreated() 
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+
+            _courseRepositoryMock.Setup(x => x.AddCourse(courseDto)).Returns(courseId);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var actualCourse = sut.AddCourse(courseDto);
+
+            //Than
+            Assert.AreEqual(courseId, actualCourse);
+            _courseRepositoryMock.Verify(x => x.AddCourse(courseDto), Times.Once());
+        }
+
+        [Test]
+        public void DeleteCourse_IntCourseId_DeleteCourse()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+
+            _courseRepositoryMock.Setup(x => x.DeleteCourse(courseId));
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            sut.DeleteCourse(courseId);
+
+            //Than
+            _courseRepositoryMock.Verify(x => x.DeleteCourse(courseId), Times.Once());
+        }
+
+        [Test]
+        public void GetCourseById_IntCourseId_ReturnCourseWithGroups()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+            _courseRepositoryMock.Setup(x => x.GetCourse(courseId)).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetCourse(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(1));
```

Comment:
please add Times in each Verify method
and please check that tose methods are not being called:
![image](https://user-images.githubusercontent.com/18329725/127709405-1b7a2564-a089-45a6-b544-e007daef45fc.png)

Reply:
+